app/scripts/migrations/127.test.ts

```diff
@@ -0,0 +1,796 @@
+    };
+
+    expect(newState.data.NetworkController).toStrictEqual(expectedState);
+  });
+
+  it('handles not well formed rpc urls', async () => {
+    const randomChainId = '0x123456';
+
+    const oldState = {
+      meta: { version: oldVersion },
+      data: {
+        NetworkController: {
+          selectedNetworkClientId: 'sepolia',
+          networkConfigurations: {
+            'network-id-1': {
+              id: 'network-id-1',
+              chainId: randomChainId,
+              nickname: 'Random Network',
+              ticker: 'FOO',
+              rpcUrl: 'http://test.endpoint/bar',
+            },
+            // This RPC url is not well formed and should be omitted
+            'network-id-2': {
+              id: 'network-id-2',
+              chainId: randomChainId,
+              nickname: 'Random Network',
+              ticker: 'FOO',
+              rpcUrl: 'not_well_formed',
+            },
+          },
+        },
+      },
+    };
+
+    const newState = await migrate(oldState);
+
+    const { networkConfigurationsByChainId } = newState.data
+      .NetworkController as {
+      networkConfigurationsByChainId: Record<
+        string,
+        {
+          defaultRpcEndpointIndex: number;
+          rpcEndpoints: {
+            url: string;
+            type: string;
+            name: string;
+            networkClientId: string;
+          }[];
+        }
+      >;
+    };
+
+    const networkConfiguration = networkConfigurationsByChainId[randomChainId];
+    expect(networkConfiguration.defaultRpcEndpointIndex).toStrictEqual(0);
+    expect(networkConfiguration.rpcEndpoints).toStrictEqual([
+      {
+        url: 'http://test.endpoint/bar',
+        type: 'custom',
+        name: 'Random Network',
+        networkClientId: 'network-id-1',
+      },
+    ]);
+  });
+
+  it('handles the case where no rpc url is well formed', async () => {
+    const randomChainId = '0x123456';
+
+    const oldState = {
+      meta: { version: oldVersion },
+      data: {
+        NetworkController: {
+          selectedNetworkClientId: 'mainnet',
+          networkConfigurations: {
+            'network-id-1': {
+              id: 'network-id-1',
+              chainId: randomChainId,
+              nickname: 'Random Network',
+              ticker: 'FOO',
+              rpcUrl: 'not_well_formed',
+            },
+            'network-id-2': {
+              id: 'network-id-2',
+              chainId: randomChainId,
+              nickname: 'Random Network',
+              ticker: 'FOO',
+              rpcUrl: 'also_not_well_formed',
+            },
+          },
+        },
+      },
+    };
+
+    const newState = await migrate(oldState);
+
+    // `randomChainId` had no well formed urls, so it should be omitted
+    expect(newState.data.NetworkController).toStrictEqual(
+      defaultPostMigrationState(),
+    );
+  });
+
+  it('handles the case where selectedNetworkClientId doesnt point to a valid endpoint', async () => {
```

Comment:
```suggestion
  it('handles the case where selectedNetworkClientId does not point to a valid network client', async () => {
```

Reply:
oh i see. Each networkConfiguration has multiple RPC Endpoints. But the RPC Endpoint has a "networkClientId" which is an artifact of the previous architecture. Do we still consider each RPC Endpoint entry a "network client"?

Reply:
That sounds right.  Stealing blurb from https://github.com/MetaMask/core/pull/4286:

`There is no longer a 1-to-1 relationship between network configuration and network client; rather, the 1-to-1 relationship exists between RPC endpoint and network client. This means that the ID of the network client which is created for an RPC endpoint is stored on that RPC endpoint instead of the whole network configuration.`